doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+key                                   = "<role name>"               ;Role configuration
+; field                               = value
+PRIMARY_GROUP                         = STRING                      ; Primary linux group
+SECONDARY_GROUP                       = STRING                      ; List of secondary linux groups
+```
+
+###  3.3 <a name='CLIYANGmodel'></a>CLI/YANG model
+
+##### YANG model
+```yang
+
+module sonic-user-mgmt {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-user-mgmt";
+	prefix user-mgmt;
+
+    description "User Management YANG Module for SONiC OS";
+
+	revision 2022-12-01 {
+        description "First Revision";
+    }
+   container sonic-user-mgmt {
+
+        container USER_TABLE {
+
+            description "USER MANAGEMENT part of config_db.json";
+
+            list USER_TABLE_LIST {
+
+                key "name";
+
+                leaf name {
+                  type string {
+                        pattern '[a-zA-Z_]([a-zA-Z0-9_-]{0,31}|[a-zA-Z0-9_-]{0,30}$)';
+                        length 1..32 {
+                              error-message "Invalid length for the username.";
+                              error-app-tag username-invalid-length;
+                        }				  
+				          }
+                  description "User name.";
+                }                
+
+                leaf state {
+                    default "enabled";
+                    type string {
+                        pattern "enabled|disabled";
+                    }
+                }
+                
+                leaf role {
+                    default "admin";
+                    type string {
+                        pattern "admin|monitor";
+                    }
+                }                
+                
+                leaf full-name {
+                    type string;
+                }
+                
+                leaf hashed-password {
+                    mandatory true;
+                    type string;
+                }
+                
+                leaf password_history {
+                    type string;
+                  	description "List of hashed passwords seperated by comma."; 
+                }
+
+                leaf passwd_date_of_last_change {
+                    type uint32;
+                }                
+               
+            } /* end of list USER_TABLE_LIST */
+
+        } /* end of container USER_TABLE */
+
+    } /* end of container sonic-user-mgmt */
+
+} /* end of module sonic-user-mgmt */
+
+
+```
+
+```yang
+//filename:  sonic-role-mgmt.yang
+module sonic-role-mgmt {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-role-mgmt";
+	prefix role-mgmt;
+
+    description "Role Management YANG Module for SONiC OS";
+
+	revision 2022-12-01 {
+        description "First Revision";
+    }
+   container sonic-role-mgmt {
+
+        container ROLE_TABLE {
```

Comment:
This table will be added to init_cfgd with admin and monitor roles only.
so we always expect to find these roles there.
If for some reason usercfgd fails to fetch this info, then it will report an error.